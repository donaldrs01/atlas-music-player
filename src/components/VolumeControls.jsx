import React from 'react';
import volume from "../assets/volume.svg";

const VolumeControls = () => {
    return (
        <div className="w-full flex flex-col space-y-3">
            <div className="flex items-center space-x-3">
                {/* Volume Button */}
                <img src={volume} className='h-6 w-6' />
                {/* Slider */}
                <input
                    type='range'
                    min="0"
                    max="100"
                    defaultValue="50"
                    className="w-full h-2 bg-gray-200 rounded-lg cursor-pointer"
                    style={{
                        accentColor: "gray",
                    }}
                />
            </div>
        </div>
    );
};

export default VolumeControls;