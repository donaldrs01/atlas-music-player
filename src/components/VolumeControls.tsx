import React from "react";
import volumeIcon from "../assets/volumeIcon.svg";

type VolumeControlsProps = {
  volume: number;
  onVolumeChange: (newVolume: number) => void;
};

const VolumeControls: React.FC<VolumeControlsProps> = ({
  volume,
  onVolumeChange,
}) => {
  // Handle volume change
  // Fetch value of HTML range element, convert to number, set state accordingly
  const handleVolumeChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    onVolumeChange(Number(event.target.value));
  };

  return (
    <div className="flex w-full flex-col space-y-3">
      <div className="flex items-center space-x-3">
        {/* Volume Button */}
        <img src={volumeIcon} className="h-6 w-6" alt="Volume" />
        {/* Slider */}
        <input
          type="range"
          min="0"
          max="100"
          value={volume}
          onChange={handleVolumeChange}
          className="h-2 w-full cursor-pointer rounded-lg bg-gray-200"
          style={{
            accentColor: "gray",
          }}
        />
      </div>
    </div>
  );
};

export default VolumeControls;
