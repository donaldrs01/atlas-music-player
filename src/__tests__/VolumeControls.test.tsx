import { render, fireEvent } from "@testing-library/react";
import { describe, test, expect, vi } from "vitest";
import VolumeControls from "../components/VolumeControls";

describe("VolumeControl Tests", () => {
  const mockOnVolumeChange = vi.fn();

  test("volume bar renders with default value", () => {
    const { container } = render(
      <VolumeControls volume={50} onVolumeChange={mockOnVolumeChange} />,
    );
    expect(container).toMatchSnapshot();
  });

  test("volume bar renders correctly with muted volume (0)", () => {
    const { container } = render(
      <VolumeControls volume={0} onVolumeChange={mockOnVolumeChange} />,
    );
    expect(container).toMatchSnapshot();
  });

  test("volume adjustment when onVolumeChange is called using slider element", () => {
    const { getAllByRole } = render(
      <VolumeControls volume={50} onVolumeChange={mockOnVolumeChange} />,
    );

    const sliders = getAllByRole("slider");
    const testslider = sliders[0]; // Pick first slider because it was picking up multiple
    fireEvent.change(testslider, { target: { value: "95" } });
    expect(mockOnVolumeChange).toHaveBeenCalledWith(95);
  });

  test("muting works as intended", () => {
    const { getAllByRole } = render(
      <VolumeControls volume={50} onVolumeChange={mockOnVolumeChange} />,
    );
    const sliders = getAllByRole("slider");
    const testslider = sliders[0];
    fireEvent.change(testslider, { target: { value: "0" } });
    expect(mockOnVolumeChange).toHaveBeenCalledWith(0);
  });
});
