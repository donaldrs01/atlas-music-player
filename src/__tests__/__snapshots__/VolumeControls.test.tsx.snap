// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VolumeControl Tests > volume bar renders correctly with muted volume (0) 1`] = `
<div>
  <div
    class="flex w-full flex-col space-y-3"
  >
    <div
      class="flex items-center space-x-3"
    >
      <img
        alt="Volume"
        class="h-6 w-6"
        src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20width='24'%20height='24'%20viewBox='0%200%2024%2024'%20fill='none'%20stroke='currentColor'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'%20class='lucide%20lucide-volume-2'%3e%3cpath%20d='M11%204.702a.705.705%200%200%200-1.203-.498L6.413%207.587A1.4%201.4%200%200%201%205.416%208H3a1%201%200%200%200-1%201v6a1%201%200%200%200%201%201h2.416a1.4%201.4%200%200%201%20.997.413l3.383%203.384A.705.705%200%200%200%2011%2019.298z'/%3e%3cpath%20d='M16%209a5%205%200%200%201%200%206'/%3e%3cpath%20d='M19.364%2018.364a9%209%200%200%200%200-12.728'/%3e%3c/svg%3e"
      />
      <input
        class="h-2 w-full cursor-pointer rounded-lg bg-gray-200"
        max="100"
        min="0"
        style="accent-color: gray;"
        type="range"
        value="0"
      />
    </div>
  </div>
</div>
`;

exports[`VolumeControl Tests > volume bar renders with default value 1`] = `
<div>
  <div
    class="flex w-full flex-col space-y-3"
  >
    <div
      class="flex items-center space-x-3"
    >
      <img
        alt="Volume"
        class="h-6 w-6"
        src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20width='24'%20height='24'%20viewBox='0%200%2024%2024'%20fill='none'%20stroke='currentColor'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'%20class='lucide%20lucide-volume-2'%3e%3cpath%20d='M11%204.702a.705.705%200%200%200-1.203-.498L6.413%207.587A1.4%201.4%200%200%201%205.416%208H3a1%201%200%200%200-1%201v6a1%201%200%200%200%201%201h2.416a1.4%201.4%200%200%201%20.997.413l3.383%203.384A.705.705%200%200%200%2011%2019.298z'/%3e%3cpath%20d='M16%209a5%205%200%200%201%200%206'/%3e%3cpath%20d='M19.364%2018.364a9%209%200%200%200%200-12.728'/%3e%3c/svg%3e"
      />
      <input
        class="h-2 w-full cursor-pointer rounded-lg bg-gray-200"
        max="100"
        min="0"
        style="accent-color: gray;"
        type="range"
        value="50"
      />
    </div>
  </div>
</div>
`;
